pkg/storage/tableland/common/sqlfactory.go

```diff
@@ -110,31 +127,68 @@ func (s *SQLFactory) SQLForInsertingRigAttributes(rigAttrTable string, rigs []lo
 	return clean(sql), nil
 }
 
+// SQLForInsertingDeals returns the SQL statement.
+func (s *SQLFactory) SQLForInsertingDeals(table string, rigs []local.Rig) (string, error) {
+	var vals [][]interface{}
+	for _, rig := range rigs {
+		for i, deal := range rig.Deals {
+			vals = append(
+				vals,
+				goqu.Vals{rig.ID, deal.DealID, deal.StorageProvider, deal.DataModelSelector, i + 1},
+			)
+		}
+	}
+	ds := s.d.Insert(table).
+		Cols("rig_id", "deal_id", "storage_provider", "data_model_selector", "deal_number").
+		Vals(vals...)
+	sql, _, err := ds.ToSQL()
+	if err != nil {
+		return "", fmt.Errorf("creating sql to insert deals: %v", err)
+	}
+	return clean(sql), nil
+}
+
```

Comment:
Transform data stored in our local SQLite db into SQL insert statements to populate our new deals table on Tableland.